pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -585,16 +586,37 @@ func (d *DependenciesDistributor) createOrUpdateAttachedBinding(attachedBinding
 				"try again later after binding is garbage collected, see https://github.com/karmada-io/karmada/issues/6034", bindingKey)
 		}
 
+		oldConflictResolution := effectiveConflictResolution(existBinding.Spec.ConflictResolution)
+		oldPreserve := ptr.Deref(existBinding.Spec.PreserveResourcesOnDeletion, false)
+		oldRequiredByLen := len(existBinding.Spec.RequiredBy)
+
+		mergedRequiredBy := mergeBindingSnapshot(existBinding.Spec.RequiredBy, attachedBinding.Spec.RequiredBy)
```

Comment:
Why should we merge requiredBy? Won't this lead to a situation where only additions occur without deletions?